### PR TITLE
Allow overriding all connection options

### PR DESCRIPTION
**Pull request by f0x52**
needed to implement https://github.com/f0x52/matrix-appservice-irc/tree/generic-connectionOpts
will probably require a version bump, which then gets updated in matrix-appservice-irc when that PR gets accepted

CI will fail, but tests were already broken before these changes

### CODE REVIEW

File: lib/irc.js [hunk 1]
```diff
@@ -87,14 +88,8 @@ function Client(server, nick, opt) {
...
-    if (typeof opt == 'object') {
-        const keys = Object.keys(opt);
-        for (let i = 0; i < keys.length; i++) {
-            var k = keys[i];
-            if (opt[k] !== undefined)
-                self.opt[k] = opt[k];
-        }
-    }
+    // Merge default options with user-supplied options
+    Object.assign(self.opt, opt)
```
Comment (Half-Shot) [inline lib/irc.js]: Are you sure this does exactly the same thing?

File: lib/irc.js [hunk 2]
```diff
@@ -856,12 +851,14 @@ Client.prototype.connect = function(retryCount, callback) {
     if (self.opt.secure) {
         connectionOpts.rejectUnauthorized = !self.opt.selfSigned;
 
-        if (typeof self.opt.secure == 'object') {
-            // copy "secure" opts to options passed to connect()
-            for (var f in self.opt.secure) {
-                connectionOpts[f] = self.opt.secure[f];
-            }
-        }
+        // if (typeof self.opt.secure == 'object') {
```
Comment (Half-Shot) [inline lib/irc.js]: Just delete rather than comment out, please :)

File: test/test-irc.js [hunk 3]
```diff
@@ -26,11 +26,12 @@ function runTests(t, isSecure, useSecureObject) {
     var client;
     if (isSecure && useSecureObject) {
         client = new irc.Client('notlocalhost', 'testbot', {
-            secure: {
-                host: 'localhost',
-                port: port,
-                rejectUnauthorized: false
-            },
+            secure: true,
+            // {
```
Comment (Half-Shot) [inline test/test-irc.js]: same here

File: lib/irc.js (hunk 1 above)
Comment (f0x52) [reply lib/irc.js]: pretty sure on that, yeah

**Comment by Half-Shot** [comment]
Sorry that we left this to rot for ages, the codebase has moved on since so we're not going to be able to merge this one cleanly. Please re-open if you think this is still a concern.

**Pull request closed without merging**